platforms/nuttx/src/px4/common/include/px4_platform/crypto.h

```diff
@@ -0,0 +1,70 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2021 Technology Innovation Institute. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+#pragma once
+
+#include <px4_platform_common/sem.h>
+#include "arch_crypto_definitions.h"
+
+/* OS specific session handle to crypto API */
+typedef class PX4CryptoSession
```

Comment:
This looks a bit strange (mix C with C++), how do you intend to use this?

Reply:
Yes, this came out strange.

I'd rather maybe wrap all the px4_encrypt_data and other functions as member functions of a class; then instead of generating a handle with "px4_crypto_open" you'd just instantiate and object out of the class

PX4Crypto crypto = new PX4Crypto(CRYPO_RSA, encryption_key, ....);
crypto.encrypt_data(bufffer, size,...);

I guess that would make much more sense than working through c-type functions and passing the handle around...?

Reply:
Yes exactly, that is pretty much what I would expect.

Reply:
Pushed another version of ulog encryption example and a version of the crypto interface, which is implemented as a pure C++ wrapper class. Looks much cleaner IMHO